BDArmory/Control/MissileFire.cs

```diff
@@ -1767,160 +1839,252 @@ void OnGUI()
                         }
                         GUIUtils.DrawTextureOnWorldPos(bombAimerPosition, texture, new Vector2(size, size), 0);
                     }
+                    //also show this for airdropped torpedoes? Have torpedo aimer circle start from surface instead torpedo prograde for airdropped toprs?
                 }
-
+                */
                 //MISSILE LOCK HUD
-                MissileBase missile = CurrentMissile;
-                if (missile)
+                //MissileBase ml = CurrentMissile;
+                if (ml)
                 {
-                    switch (missile.TargetingMode)
+                    if (showBombAimer)
                     {
-                        case MissileBase.TargetingModes.Laser:
-                            {
-                                if (laserPointDetected && foundCam)
-                                {
-                                    GUIUtils.DrawTextureOnWorldPos(foundCam.groundTargetPosition, BDArmorySetup.Instance.greenCircleTexture, new Vector2(48, 48), 1);
-                                }
-                                else
+                        MissileLauncher msl = CurrentMissile as MissileLauncher;
+                            //boreRadarRing.SetActive(false);
+                            boreRing.SetActive(true);
+                            Quaternion rotation = Quaternion.LookRotation(bombAimerTerrainNormal, boreRing.transform.forward);
+                            boreRing.transform.SetPositionAndRotation(bombAimerPosition + VectorUtils.GetUpDirection(bombAimerPosition) * 5, rotation);
+                        if (vessel.altitude > msl.GetBlastRadius())
+                        {
+                            if (guardTarget && (foundCam && (foundCam.groundTargetPosition - guardTarget.transform.position).sqrMagnitude <= 100))
+                                boreRing.transform.localScale = Vector3.one * 50;
+                            else
+                                boreRing.transform.localScale = Vector3.one * Mathf.Min(150, msl.GetBlastRadius()) / 10; //ring model has 10m radius.
+                        }
+                        else 
+                        {
+                            GUIUtils.DrawTextureOnWorldPos(bombAimerPosition, BDArmorySetup.Instance.greenSpikedPointCircleTexture, new Vector2(128, 128), 0);
+                        }
+                        if (guardTarget) GUIUtils.DrawTextureOnWorldPos(AIUtils.PredictPosition(guardTarget, bombFlightTime, immediate: false), BDArmorySetup.Instance.greenDotTexture, new Vector2(6, 6), 3);
+                        else GUIUtils.DrawTextureOnWorldPos(bombAimerPosition, BDArmorySetup.Instance.greenDotTexture, new Vector2(6, 6), 0);
+                        //The dynamic bore ring works very well for the bomb aimer. Less convinced about dynamicly sized missile FOV aimers. Ah, well. Easy enough to revert back to the DrawTexture method
```

Comment:
Checking (since it doesn't do that on my end), I didn't upload the newer version of the ring, which means you've got the original version that's not properly oriented.

Reply:
Testing with the latest changes, I think it'd be better without being adjusted for the terrain normals and just have it be a circle with the axis pointing at the camera (i.e., like the outline of a sphere).

Reply:
moved and reoriented the ring to be pointed at the camera